toolchain/gcc.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash -xe
+# Script for building a RISC-V GNU Toolchain from checked out sources
+
+# Copyright (C) 2020 Embecosm Limited
+
+# Contributor: Simon Cook <simon.cook@embecosm.com>
+
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# Variables used in this script
+INSTALLPREFIX=${PWD}/install
+BUILDPREFIX=${PWD}/build
+SRCPREFIX=${PWD}
+DEFAULTARCH=rv32imac
+DEFAULTABI=ilp32
+
+# Print the GCC and G++ used in this build
+which gcc
+which g++
+
+# If a BUGURL and PKGVERS has been provided, set variables
+EXTRA_OPTS=""
+if [ "x${BUGURL}" != "x" ]; then
+  EXTRA_OPTS="${EXTRA_OPTS} --with-bugurl='${BUGURL}'"
+fi
+if [ "x${PKGVERS}" != "x" ]; then
+  EXTRA_OPTS="${EXTRA_OPTS} --with-pkgversion='${PKGVERS}'"
+fi
+
+# Allow environment to control parallelism
+
+PARALLEL_JOBS=$(nproc)
+
+# If a local GMP build is not available, download and build it
+source utils/prepare-libgmp.sh
```

Comment:
This would fails if we have the -e flag set as that file does not exist in our repo, thus we dont build GMP, It appears to be hard requirement for GDB, which is fine I guess since we dont use it !

Reply:
Definitely, I encountered this issue when trying out the script yesterday.
Worked around it by commenting out this line from the script, but it should be fixed appropriately.

Build seems to still work without this line (ran on a CentOS 7 WSL distro - yay legacy software!) and it at least gave reasonable output from `gcc -v`, so it might be fine like this.